docs: update build provider procedures (#5537)
canonical/snapcraft: docs/how-to/setup/set-up-snapcraft.rst

    First, `install Brew <https://brew.sh#install>`_.

    Then, install Snapcraft through Brew:

    .. code:: bash

        brew install snapcraft

  .. group-tab:: Windows

    First, `install WSL2 with Ubuntu 20.04 or higher
    <https://learn.microsoft.com/en-us/windows/wsl/install#install-wsl-command>`_.

    Then, install Snapcraft in WSL2:

    .. code:: bash

        snap install snapcraft --classic

Install a build provider
~~~~~~~~~~~~~~~~~~~~~~~~

Snapcraft relies on a *build provider* to create an isolated build environment, like a
sandbox. Inside this environment, software can be built and packaged as snaps without
making potentially destructive changes to the host system.

.. list-table::
    :widths: 1 3 2

    * - Build provider
      - Description
      - Default on
    * - `LXD <https://canonical.com/lxd>`_
      - Creates and manages Linux container images. It can operate inside VMs.
      - Snapcraft 7 and higher on Linux
    * - `Multipass <https://multipass.run>`_
      - Creates and manages virtual machine (VM) build instances. It automates setup and
        teardown of cloud-style Ubuntu VMs. It can't be used  reliably on platforms that

>         don't support nested virtualization. For instance, it most likely won't itself
>         run inside a VM.

Reviewer: ```suggestion
        don't support nested virtualization. For instance, it most likely won't
        run inside a VM.
```

Author: I reworded the sentence. c35814f

      - - Snapcraft 6 on Linux
        - All versions on macOS and Windows

> When you first run Snapcraft, it installs the default build provider. If the default

Reviewer: ```suggestion
When you first run Snapcraft, it installs the default build provider for your host system. If the default
```

Author: c35814f

build provider isn't suitable, you can install the alternative and then :ref:`switch to
it <how-to-select-a-build-provider>`.

.. _how-to-set-up-snapcraft-install-lxd:

Install LXD
^^^^^^^^^^^

Install the LXD snap:

.. code:: bash

    snap install lxd

LXD has special requirements from your local user account that Snapcraft can't manage
for you during its automatic setup. Add your user account to the ``lxd`` group so you
can access the LXD daemon:

.. code:: bash

    sudo usermod -a -G lxd $USER

> Log out and back in to your account on your system for the new group to take effect.

Reviewer: ```suggestion
Log out and back in to your local user account for the new group to take effect.
```
There may be some nuance I'm missing here, but this seems more consistent with the previous paragraph.

Author: It was an attempt to avoid confusion with signing in to Snapcraft itself. c35814f

        brew install snapcraft

  .. group-tab:: Windows

    First, `install WSL2 with Ubuntu 20.04 or higher
    <https://learn.microsoft.com/en-us/windows/wsl/install#install-wsl-command>`_.

    Then, install Snapcraft in WSL2:

    .. code:: bash

        snap install snapcraft --classic

Install a build provider
~~~~~~~~~~~~~~~~~~~~~~~~

Snapcraft relies on a *build provider* to create an isolated build environment, like a
sandbox. Inside this environment, software can be built and packaged as snaps without
making potentially destructive changes to the host system.

.. list-table::
    :widths: 1 3 2

    * - Build provider
      - Description
      - Default on
    * - `LXD <https://canonical.com/lxd>`_
      - Creates and manages Linux container images. It can operate inside VMs.
      - Snapcraft 7 on Linux
    * - `Multipass <https://multipass.run>`_
      - Creates and manages virtual machine (VM) build instances. It automates setup and
        teardown of cloud-style Ubuntu VMs. It can't be used  reliably on platforms that
        don't support nested virtualization. For instance, it most likely won't itself
        run inside a VM.
      - Snapcraft 6 on Linux

>         macOS and Windows

Author: Until core20 is done and dusted, I see no reason not to mention the difference. It certainly took me by surprise during testing.

When you first run Snapcraft, it installs the default build provider. If the default

> build provider isn't fit for purpose, you can install the alternative and then

Reviewer 2: ```suggestion
build provider isn't fit for your purpose, you can install the alternative and then
```

Author: "fit for purpose" is a fixed phrase in English, but it's maybe too fancy. [0c39d3c](https://github.com/canonical/snapcraft/pull/5537/commits/0c39d3c4bd696fd5b1c0e10dceae39fae22e58c7)

Reviewer 2: Ha, certainly too fancy for me to recognize!
